PYME/Acquire/SpoolController.py

```diff
@@ -482,13 +522,43 @@ def stop_spooling(self):
         return 'OK'
 
     @webframework.register_endpoint('/start_spooling', output_is_json=False)
-    def start_spooling(self, filename=None, max_frames=sys.maxsize):
-        self.spool_controller.StartSpooling(fn=filename, maxFrames=max_frames)
+    def start_spooling(self, body='{}'):
+        settings = json.loads(body)
+        self.spool_controller.StartSpooling(**settings)
         return 'OK'
+
+
+class SpoolControllerServer(webframework.APIHTTPServer, SpoolControllerWrapper):
```

Comment:
There should be no need for recipes to interact directly with the spool controller. This is super unsafe in terms of race conditions. The one and only place where recipes should interact with the microscope is through queueing actions.